### PR TITLE
Fixed bugs and combined prediction/average api

### PR DESCRIPTION
**Pull request by jsong336**

### CODE REVIEW

File: _server/api/rental.py [hunk 1]
```diff
@@ -32,95 +33,61 @@ def get_rentals():
...
 
-        # Predict.
-        predictions.append(rf_model.predict(x).item())
+    if any(invalid_args):
+        raise BadRequest('Must provide universityId and postalCode')
 
-    return {
-        'prediction': np.mean(predictions)
-    }
+    """ Filter baed average starts here by default. """
```
Comment (hPark0811) [inline _server/api/rental.py:78]: baed => based

File: _server/api/rental.py [hunk 2]
```diff
@@ -132,35 +99,127 @@ def get_average_rental():
...
-        RentalRange.rentToUniversityDistance >= min_distance_km
+        RentalRange.rentToUniversityDistance >= min_distance
     ).filter(
-        RentalRange.rentToUniversityDistance <= max_distance_km
-    )
+        RentalRange.rentToUniversityDistance <= max_distance
+    ) 
 
-    if property_types:
+    # Filter baed on propertyType, bathCount, bedCount. If none, automatically unfiltered.
```
Comment (hPark0811) [inline _server/api/rental.py:107]: baed => based

File: _server/requirements.txt [hunk 3]
```diff
@@ -30,7 +30,7 @@ PyMySQL==0.10.0
 python-dateutil==2.8.1
 pytz==2020.1
 requests==2.24.0
-scikit-learn==0.23.2
+scikit-learn==0.22.2.post1
```
Comment (hPark0811) [inline _server/requirements.txt:33]: why did we downgrade version to 22.2 post version here?

File: _client/livethere-app/src/components/summary/rentalSummary/RentalSummary.js [hunk 4]
```diff
@@ -177,40 +143,34 @@ const RentalSummary = (props) => {
...
+    summaryText = (
       <div>
-        {
-          prediction ? (
-            <div>
-              Predicted rental price is <b>${prediction.prediction?.toFixed(0)}/mo per room</b>
-            </div>
-          ) : null
-        }
+          Predicted rental price is <b>${summary.summary?.toFixed(0)}/month per room</b>
```
Comment (hPark0811) [inline _client/livethere-app/src/components/summary/rentalSummary/RentalSummary.js:163]: can we rename the response from backend so it's not summary.summary? maybe summary.estimate or why not just keep it the old way like summary.average?

File: _server/api/rental.py [hunk 5]
```diff
@@ -132,35 +99,127 @@ def get_average_rental():
...
+
+    # Check if prediction is available.
+    prd_valid = [
+        request.args.get('propertyType') is not None,
+        # request.args.get('bedCount') is not None,
+        # request.args.get('bathCount') is not None
+    ]
+    
+    """ Prediction starts here """    
+    if (not all(valid_avg)) and all(prd_valid):
```
Comment (hPark0811) [inline _server/api/rental.py:149]: not all(valid_avg)) doesn't need to be checked again, as we're already returning something from previous condition.

File: _client/livethere-app/src/components/summary/rentalSummary/RentalSummary.js [hunk 6]
```diff
@@ -10,32 +10,29 @@ import Grid from "@material-ui/core/Grid";
...
 
   const fetchRentalSummary = () => {
-    console.log('fetch rental summary data');
-
-    let params = {universityId: props.universityId};
-
-    if (maxDistance) {
+    let params = {
+      universityId: props.universityId,
+      postalCode: props.postalCode.slice(0, 3) + " " + props.postalCode.slice(3)
```
Comment (hPark0811) [inline _client/livethere-app/src/components/summary/rentalSummary/RentalSummary.js:30]: I think this is a pretty sketchy way of doing it. Can we sanitize data in the backend instead? This is just a recommendation.

File: _server/requirements.txt (hunk 3 above)
Comment (jsong336) [reply _server/requirements.txt:33]: ML model is in 22.2.post

File: _client/livethere-app/src/components/summary/rentalSummary/RentalSummary.js (hunk 4 above)
Comment (jsong336) [reply _client/livethere-app/src/components/summary/rentalSummary/RentalSummary.js:163]: Ill rename it to value. Summary.summary is either average or prediction

File: _client/livethere-app/src/components/summary/rentalSummary/RentalSummary.js (hunk 6 above)
Comment (jsong336) [reply _client/livethere-app/src/components/summary/rentalSummary/RentalSummary.js:30]: Alright

File: _server/api/rental.py (hunk 5 above)
Comment (jsong336) [reply _server/api/rental.py:149]: Right

**Review by hPark0811 — DISMISSED** [review]

**Pull request closed without merging**